hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -154,7 +156,18 @@ protected Object findProxied(HibernateProxy proxy)
                     if (session != null) {
                         idName = session.getFactory().getIdentifierPropertyName(init.getEntityName());
                     } else {
-                        idName = init.getEntityName();
+                        String idFieldName = null;
+                        try {
+                            for (Field field : init.getPersistentClass().getDeclaredFields()) {
+                                if (field.isAnnotationPresent(Id.class)) {
+                                    idFieldName = field.getName();
+                                    break;
+                                }
+                            }
+                        } catch (Exception e) {
+                            e.printStackTrace();
```

Comment:
I fully agree with you. I just wanted to say, that it would be great if you could add this functional to your library. And you as a developer should know how to implement this functionality.